server/src/main/java/com/tencent/rss/server/HealthCheck.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Tencent is pleased to support the open source community by making
+ * Firestorm-Spark remote shuffle server available.
+ *
+ * Copyright (C) 2021 THL A29 Limited, a Tencent company.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * https://opensource.org/licenses/Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.tencent.rss.server;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Lists;
+import com.google.common.util.concurrent.Uninterruptibles;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * HealthCheck will check every server whether has the ability to process shuffle data. Currently, we only support disk
+ * checker. If enough disks don't have enough disk space, server will become unhealthy, and only enough disks
+ * have enough disk space, server will become healthy again.
+ **/
```

Comment:
As comments said, disk check is one of healthCheck, how about create a Checker interface?
DiskChecker implements Checker
Shuffle server register checker to healthCheck

Reply:
It's a good idea. I can add a `todo`.But I think it is a little complex. We can implement it in another pr.

Reply:
It's better to be implemented in the first step, and not in a refactor PR.

Reply:
OK.